src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const [user, setUser] = useState<User | null>(null);
+
+  if (!currentTodo) {
+    throw Error('No current todo selected.');
+  }
```

Comment:
Throwing an error when `currentTodo` is not selected can lead to an application crash. Consider handling this scenario more gracefully, perhaps by rendering a fallback UI or redirecting the user to a safe state.